### PR TITLE
parallel pylint execution

### PR DESCRIPTION
**Pull request by 1ucian0**
Recently, @atilag showed concerns about the project execution time in Travis. This PR parallelizes `pylint` execution of the two modules, `qiskit` and `test`.

Locally, this creates some gains in time:
```
$ time pylint -rn qiskit test
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

real	2m35.476s
user	2m24.476s
sys	0m5.903s
```
Compare with:
```
$ time pylint -j 2 -rn qiskit test
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

real	1m43.819s
user	3m5.807s
sys	0m11.633s
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,7 +38,7 @@ env:
 
 # Ignoring generated ones with .py extension.
 lint:
-	pylint -rn qiskit test
+	pylint -j 2 -rn qiskit test
```
Comment (kdk) [inline Makefile]: Could use `-j 0` (pylint will auto-detect CPU count)?

Comment (kdk) [reply Makefile]: Also, looks like there are failures for missing monkey-patched gates (presumably due to changing the evaluation order, `pylint test/ qiskit/` also raises these.)

**Comment by 1ucian0** [comment]
I'm getting longer time with `-j 0`. I think is because [each subprocess gets one module](https://docs.pylint.org/en/1.6.0/run.html#parallel-execution).

File: Makefile (hunk 1 above)
Comment (1ucian0) [reply Makefile]: Is `ignored-classes=QuantumCircuit` too much? I can clean up all the around ` # pylint: disable=no-member`

**Comment by kdk** [comment]
> I'm getting longer time with `-j 0`. I think is because [each subprocess gets one module](https://docs.pylint.org/en/1.6.0/run.html#parallel-execution).

It's ~slightly faster for me, but not significantly. And it does seem to spin up 8 processes, so maybe its split by source module (and not by command line argument).

```
$ time pylint -rn -j 1 qiskit/ test/

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)


real	2m51.551s
user	2m46.864s
sys	0m4.284s

$ time pylint -rn -j 2 qiskit/ test/
...
<~ dozen no-member failures>
...
-------------------------------------------------------------------
Your code has been rated at 9.97/10 (previous run: 10.00/10, -0.02)


real	1m56.668s
user	3m41.570s
sys	0m9.866s

$ time pylint -rn -j 0 qiskit/ test/
...
<~ several hundred no-member failures>
...
-----------------------------------
Your code has been rated at 9.03/10


real	1m19.318s
user	9m22.396s
sys	0m25.218s
```

File: Makefile (hunk 1 above)
Comment (kdk) [reply Makefile]: `ignored-classes=QuantumCircuit` seems reasonable, from the doc, QuantumCircuit fits its purpose well:
```
    --ignored-classes=<members names>
                        List of class names for which member attributes should
                        not be checked (useful for classes with dynamically
                        set attributes). This supports the use of qualified
                        names. [current:
                        optparse.Values,thread._local,_thread._local]
```

**Pull request merged**